### PR TITLE
Add state visibility between Executions and Operations.

### PR DESCRIPTION
**Pull request by cschanck**
Added a state variable in Operation, which tracks the state of the
Execution currently running. In the Execution implementations, before
starting, the state for all the operations in the scenario is set
to BEGINNING. Upon completion (if an error) or just prior to shutting
down the Execution, the state for all the Operations is changed to
ENDING.

This allows for user implemented operations to catch RuntimeExceptions
(which may be caused by thread interrupts, etc) and discard them if
the Operation is already shutting down.

Additionally, a small change was made in HtmlReporter. The Histogram
was being arbitrarily corrected by 1000000L; in certain cases this
cases an IndexOutOfBounds exception, or other weird exceptions. In this
case, the historgram is generated uncorrected. Anyone with a better idea
is welcome to contribute.

### CODE REVIEW

File: rainfall-core.iml [hunk 1]
```diff
@@ -0,0 +1,227 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aurbroszniowski) [inline rainfall-core.iml]: is there a good reason to include iml?

File: src/main/java/io/rainfall/Scenario.java [hunk 2]
```diff
@@ -1,56 +1,56 @@
...
-  public static Scenario scenario(final String name) {
-    return new Scenario(name);
-  }
-
-  public List<RangeMap<Operation>> getOperations() {
-    return this.operations;
-  }
-
-}
+/*
```
Comment (aurbroszniowski) [inline src/main/java/io/rainfall/Scenario.java]: it seems to me that it's only reformatting or am I missing something?

File: src/main/java/io/rainfall/execution/Executions.java [hunk 3]
```diff
@@ -1,55 +1,55 @@
...
-
-  public static Ramp ramp(From from, To to, Every every, Over over) {
-    return new Ramp(from, to, every, over);
-  }
-
-  public static RunsDuring during(int nb, TimeDivision timeDivision) {
-    return new RunsDuring(nb, timeDivision);
-  }
-}
+/*
```
Comment (aurbroszniowski) [inline src/main/java/io/rainfall/execution/Executions.java]: same as above

File: src/main/java/io/rainfall/execution/InParallel.java [hunk 4]
```diff
@@ -61,7 +61,7 @@ public InParallel(final int nb, final Unit unit, final Every every, final TimeMe
   public <E extends Enum<E>> void execute(final StatisticsHolder<E> statisticsHolder, final Scenario scenario,
                                           final Map<Class<? extends Configuration>, Configuration> configurations,
                                           final List<AssertionEvaluator> assertions) throws TestException {
-    final ConcurrencyConfig concurrencyConfig = (ConcurrencyConfig)configurations.get(ConcurrencyConfig.class);
+    final ConcurrencyConfig concurrencyConfig = (ConcurrencyConfig) configurations.get(ConcurrencyConfig.class);
```
Comment (aurbroszniowski) [inline src/main/java/io/rainfall/execution/InParallel.java]: so it seems that a number of reformatting are in that PR, maybe I missed something but usually it's better to reformat only code that was changed, or if reformatting is important for some specific purpose, make it part of a specific PR

File: rainfall-core.iml (hunk 1 above)
Comment (cschanck) [inline rainfall-core.iml]: No, that can go.

File: src/main/java/io/rainfall/Scenario.java (hunk 2 above)
Comment (cschanck) [inline src/main/java/io/rainfall/Scenario.java]: No, was just formatting. Sorry!

File: src/main/java/io/rainfall/execution/Executions.java (hunk 3 above)
Comment (cschanck) [inline src/main/java/io/rainfall/execution/Executions.java]: Yes

File: src/main/java/io/rainfall/execution/InParallel.java (hunk 4 above)
Comment (cschanck) [inline src/main/java/io/rainfall/execution/InParallel.java]: Yeah, I must have hit alt-f10 at some point and reformatted by accident.

**Pull request merged**